meta-mender-core/classes/mender-setup.bbclass

```diff
@@ -62,11 +62,27 @@ MENDER_DATA_PART_DEFAULT = "${MENDER_STORAGE_DEVICE_BASE}${MENDER_DATA_PART_NUMB
 MENDER_MTD_UBI_DEVICE_NAME ??= "${MENDER_MTD_UBI_DEVICE_NAME_DEFAULT}"
 MENDER_MTD_UBI_DEVICE_NAME_DEFAULT = ""
 
-# Filesystem type of data partition. This configuration is used in fstab. Most
-# filesystems can be auto detected, but some can not and hence we allow the
-# user to override this.
+# Filesystem type of data partition. Used for both FS generation and fstab construction
+# Leave as default (auto) to generate a partition using the same Filesystem as 
+# the rootfs ($ARTIFACTIMG_FSTYPE) and set the fstab to auto-detect the partition type
+# Set to a known filesystem to generate the partition using that type
 MENDER_DATA_PART_FSTYPE ??= "${MENDER_DATA_PART_FSTYPE_DEFAULT}"
 MENDER_DATA_PART_FSTYPE_DEFAULT = "auto"
+
+# Filesystem type of data partition to generate. Used only for FS generation
+# Typically you'll be best off setting MENDER_DATA_PART_FSTYPE instead
+MENDER_DATA_PART_FSTYPE_TO_GEN ??= "${@bb.utils.contains('MENDER_DATA_PART_FSTYPE', 'auto', '${ARTIFACTIMG_FSTYPE}', '${MENDER_DATA_PART_FSTYPE}', d)}"
```

Comment:
Use the `DEFAULT` pattern that the other variables use, please.

The reason we do it this way is because:

```
MENDER_DATA_PART_FSTYPE_TO_GEN ??= "wrong"
MENDER_DATA_PART_FSTYPE_TO_GEN_arm ??= "also-wrong"
MENDER_DATA_PART_FSTYPE_TO_GEN = "right"

MENDER_DATA_PART_FSTYPE_TO_GEN evaluates to "also-wrong"
```

However, with the `DEFAULT` pattern:

```
MENDER_DATA_PART_FSTYPE_TO_GEN ??= "${MENDER_DATA_PART_FSTYPE_TO_GEN_DEFAULT}"
MENDER_DATA_PART_FSTYPE_TO_GEN_DEFAULT = "wrong"
MENDER_DATA_PART_FSTYPE_TO_GEN_DEFAULT_arm = "also-wrong"
MENDER_DATA_PART_FSTYPE_TO_GEN = "right"

MENDER_DATA_PART_FSTYPE_TO_GEN evaluates to "right"
```

Reply:
I'm not sure I completely grasp why it works like that, (and I'm a bit confused because I don't see any suffixes on the DEFAULTS in mender-setup) but I'll make the change.

Reply:
Oh, wait, it wasn't that hard, it's just that the override takes precedence and using the second version is just prep for when we'd want to set the default for certain overrides, even if we're not doing so yet.